### PR TITLE
Replaced usage of where.delete/destroy_all with delete/destroy_by

### PR DESCRIPTION
**Pull request by abhaynikam**
Follow up on #35316.

Replaced the usage of `where().delete/destroy_all` with `delete/destroy_by`.

### CODE REVIEW

File: activerecord/test/models/topic.rb [hunk 1]
```diff
@@ -99,7 +99,7 @@ def default_written_on
     end
 
     def destroy_children
-      self.class.where("parent_id = #{id}").delete_all
+      self.class.delete_by("parent_id = #{id}")
```
Comment (abhaynikam) [inline activerecord/test/models/topic.rb]: I am not sure why we have used string interpolation here. `"parent_id = #{id}"` can be replaced with `parent_id: id`

Comment (kaspth) [reply activerecord/test/models/topic.rb]: Yeah, that does seem odd. Does git blame or the Topic using tests say anything? While it doesn't matter in tests, this also technically is vulnerable to SQL injections so at the very least it should use `= ?`. But let's just switch it to the hash as you mention.

Comment (abhaynikam) [reply activerecord/test/models/topic.rb]: @kaspth Changes seem to be very old can be changed to `parent_id: id`. I have done that here.

File: guides/source/active_record_basics.md [hunk 2]
```diff
@@ -314,7 +314,7 @@ method:
 
 ```ruby
 # find and delete all users named David
-User.where(name: 'David').destroy_all
+User.destroy_by(name: 'David')
```
Comment (matthewd) [inline guides/source/active_record_basics.md]: > If you'd like to delete several records in bulk, you may use `destroy_all` method

**Pull request merged**